serveur-web/api-v2/model/bucket.py

```diff
@@ -16,29 +16,77 @@
 import json
 from database.database import Database
 
-class BucketModel(object):
+class Bucket(object):
 
-	@staticmethod
-	def find_by_id(id):
+	def __init__(self):
+		self.id = 0
+		self.id_plant = 0
+		self.name = ""
+		self.ip_address = ""
+
+
+	def __init__(self, id, id_plant, name, ip_address):
+		self.id = id
+		self.id_plant = id_plant
+		self.name = name
+		self.ip_address = ip_address
+
+
+	def to_detailed_json(self):
+		serialized = {
+				"id" : self.id,
+				"id_plant" : self.id_plant,
+				"name" : self.name,
+				"ip_address" : self.ip_address
+			}
+
+		return serialized
+
+
+	def transform(bucket_data):
+		bucket = Bucket(bucket_data[0],
+				bucket_data[1],
+				bucket_data[2],
+				bucket_data[3])
+
+		return bucket
+
+
+	def save(self):
+		"""
+		Create or update a bucket
+		"""
+		if self.id is None:
+			return Bucket.create(self)
+		else:
+			Bucket.update(self)
+			return self
+
+	"""
+	Static Methods
+	"""
+
+	@classmethod
+	def get(cls,id):
 		"""
 		Retreive a bucket's data and return it
 		"""
 		db = Database.get_instance()
 		bucket_data = db.execute("SELECT * FROM BUCKET WHERE ID='"+str(id)+"'")
 		if bucket_data:
-			return bucket_data[0]
+			return cls.transform(bucket_data[0])
 		else:
 			return None
 
-	@staticmethod
-	def get_all():
+	@classmethod
+	def get_all(cls):
 		"""
 		Retreive a bucket's data and return it
 		"""
 		db = Database.get_instance()
 		bucket_data = db.execute("SELECT * FROM BUCKET")
 		if bucket_data:
-			return bucket_data
+			return [cls.transform(b) for b in bucket_data]
```

Comment:
Suggestion : renvoyer simplement « return [cls.transform(b) for b in bucket_data] ». Comme ça on renvoie toujours une liste valide. Au pire, on teste si elle est vide plutôt que si elle est None. Dans tous les cas, on sauve une condition.